plugins/rebalance/rest.go

```diff
@@ -0,0 +1,235 @@
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	// collect inputs from url
+	volname := mux.Vars(r)["volname"]
+
+	var req rebalanceapi.StartReq
+
+	// Unmarshal Request so to handle fix-layout and start force
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusUnprocessableEntity, "Unable to parse the request", api.ErrCodeDefault)
+		return
+	}
+
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusNotFound, "Invalid volume", api.ErrCodeDefault)
+		return
+	}
+
+	if vol.State != volume.VolStarted {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume not started", api.ErrCodeDefault)
+		return
+	}
+
+	if vol.DistCount == 1 {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume is not distributed volume",
+			api.ErrCodeDefault)
+		return
+	}
+
+	// Check for remove- brick pending
+	//TODO
+
+	// A simple transaction to start rebalance on all nodes
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+	lock, unlock, err := transaction.CreateLockSteps(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Nodes = vol.Nodes()
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "rebalance-start.Commit",
+			Nodes:  txn.Nodes,
+		},
+		{
+			DoFunc: "rebalance.StoreVolume",
+			Nodes:  []uuid.UUID{gdctx.MyUUID},
+		},
+		unlock,
+	}
+
+	err = txn.Ctx.Set("volname", volname)
+	if err != nil {
+		logger.WithError(err).Error("failed to set request in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	rebal := createRebalanceInfo(volname, &req)
+	if rebal == nil {
+		logger.WithError(err).Error("failed to create Rebalance info")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	err = txn.Ctx.Set("rinfo", rebal)
+	if err != nil {
+		logger.WithError(err).Error("failed to set rebalance info in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	err = txn.Do()
+	if err != nil {
+		/* TODO: Need to handle failure case. Unlike other process,
+		 * rebalance process is one per node and depends on volfile change.
+		 * Need to handle scenarios where process is started in
+		 * few nodes and failed in few others */
+		logger.WithFields(log.Fields{
+			"error":   err.Error(),
+			"volname": volname,
+		}).Error("failed to start rebalance on volume")
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Ctx.Logger().WithField("volname", rebal.Volname).Info("rebalance started")
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, rebal)
```

Comment:
Is the contents of `rebal` variable supposed to be stored in the store ?
Or is there supposed to be a call to `storeRebalanceDetails()` somewhere in this handler ?